include/simdjson/inline/document_stream.h

```diff
@@ -5,10 +5,70 @@
 #include <algorithm>
 #include <limits>
 #include <stdexcept>
-
 namespace simdjson {
 namespace dom {
 
+#ifdef SIMDJSON_THREADS_ENABLED
+namespace {
+  // Putting those inside the stage1_worker class may create problems with move constructors
+  // since a mutex is not supposed to be moved.
```

Comment:
The fact that multiple simultaneous parse_manys might block on each other is worth noting, though.

Reply:
Damn. Good catch.

I don't know how solve this one.

Reply:
I could imagine making document_stream unmoveable--or better, make parser unmoveable and store the worker there.

Reply:
Where do I put the mutex given that a mutex cannot be moved?

Reply:
Ah. Ok. I could use a unique pointer.

Reply:
Storing the worker there would mean we keep it alive (but maybe sleeping?) until the parser goes away, which would be good for multiple parse_many calls ...